pkg/network/node/pod.go

```diff
@@ -192,42 +184,46 @@ func (m *podManager) Start(rundir string, localSubnetCIDR string, clusterNetwork
 	return m.cniServer.Start(m.handleCNIRequest)
 }
 
-func (m *podManager) InitRunningPods(existingSandboxPods map[string]*kruntimeapi.PodSandbox, existingOFPodNetworks map[string]podNetworkInfo, cRunningPods []kapi.Pod) error {
+func (m *podManager) InitRunningPods(existingSandboxPods map[string]*kruntimeapi.PodSandbox, existingOFPodNetworks map[string]podNetworkInfo, cRunningPods map[string]kapi.Pod, clearHostPorts bool) error {
 	m.runningPodsLock.Lock()
 	defer m.runningPodsLock.Unlock()
-
-	for _, cPod := range cRunningPods {
-		cKey := getPodKey(cPod.Namespace, cPod.Name)
-
+	if m.enableHostports {
+		iptInterface := utiliptables.New(utilexec.New(), utildbus.New(), utiliptables.ProtocolIpv4)
+		m.hostportSyncer = kubehostport.NewHostportSyncer(iptInterface)
```

Comment:
seems strange to me that you have to initialize `m.hostportSyncer` twice; is that correct?

Reply:
Essentially this block of code was executed in `Start`. However we never initialized state for any running pods in that function. Seeing as how  `InitRunningPods`...initializes running pods, and, in particular, now, their host ports: I moved it here. Thus, in case we need to clear the host ports, we do so here and then initialize the host ports right after for any pods needing such.